### ISSUE TITLE
Blocks broken by natural sources are not traced

### ISSUE THREAD

**Issue by OmerBenGera** [BG-Software-LLC/SuperiorSkyblock2#1014]
### Minecraft's Version

All minecraft versions

### Plugin's Version

1.9.0.570

### Describe the bug

When a block is broken by a natural source, it will not get detected as a broken block.
An example for such event is cactus being broken when it's bottom cactus is broken.

### To Reproduce

Make a block get broken naturally and it won't be detected.

**Comment by DebuffPvP** [comment]
Is this in any way related to mining in general? So i made a plugin so that when a block of sorts defoned in the config is broken the player has a chance of earning one of the given rewards in the config at a set chance! But the only issue is that when  blocks are broken and they place it again they can remine it to receive a reward possibly unless blocks are logged i believe is what the issue is but i've not come across a solution since minecraft has too many blocks and logging all the miners might be an issue.

**Comment by OmerBenGera** [comment]
> Is this in any way related to mining in general? So i made a plugin so that when a block of sorts defoned in the config is broken the player has a chance of earning one of the given rewards in the config at a set chance! But the only issue is that when blocks are broken and they place it again they can remine it to receive a reward possibly unless blocks are logged i believe is what the issue is but i've not come across a solution since minecraft has too many blocks and logging all the miners might be an issue.

I dont understand, however doesnt sound like its related to this issue

**Comment by OmerBenGera** [comment]
Should be fixed in Paper - still an issue with Spigot tho.

**Comment by Lemvs** [comment]
@OmerBenGera I'm using `Paper version 1.21.11-131-ver/1.21.11@6d5b910 (2026-05-03T16:08:07Z) (Implementing API version 1.21.11-R0.1-SNAPSHOT)` with `SuperiorSkyblock2 2026.1-b859` and the problem you're writing about in this issue (cactus being broken by block/block below cactus was broken is not detected) still exists, so either something has changed or something in the code has been messed up, because it doesn't work as it should, even though it's Paper.

**Comment by Lemvs** [comment]
Yes, you removed that listener, which is why there is now a problem with cactus:
https://github.com/BG-Software-LLC/SuperiorSkyblock2/commit/fccd108dee79692303daaa0b8c16da9116b4bc4b
